pkg/lib/theme/theme.go

```diff
@@ -197,30 +171,51 @@ func collectTokensAsString(tokens []css.Token) string {
 	return buf.String()
 }
 
-func transform(elements []element) (out []element) {
+func stringify(buf *bytes.Buffer, elements []element) {
+	for _, element := range elements {
+		var indent indentation
+		element.Stringify(buf, indent)
+	}
+}
+
+func parseCSSRawString(cssStr string) ([]element, error) {
+	b := []byte(cssStr)
+	r := bytes.NewReader(b)
+	p := css.NewParser(parse.NewInput(r), false)
+	var elements []element
+	for {
+		var el element
+		el, err := parseElement(p)
+		if errors.Is(err, io.EOF) {
+			err = nil
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+		elements = append(elements, el)
+	}
+	return elements, nil
+}
+
+// CheckDeclarationInSelector checks if the declaration is in the selector of provided css
+func CheckDeclarationInSelector(cssString string, selector string, declarationProperty string) (bool, error) {
```

Comment:
Add `func SetDeclarationInSelector(cssString string, selector string, declaration declaration) (newCSSString, outcome outcome)`

Reply:
[Per this comment](https://github.com/authgear/authgear-server/pull/4776#discussion_r1785608930), will add below instead

```go
// If added, then added is true.
AddDeclarationInSelectorIfNotPresentAlready(cssString string, selector string, declaration declaration) (newCSS string, added bool). 
```